frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -300,7 +300,8 @@ export default ({
               ...createMessage({ meta, data, hash: message.hash() }),
               // TODO: pending is useful to turn the message gray meaning failed (just like Slack)
               // when we don't get event after a certain period
-              pending: true
+              pending: true, // NOTE: do not RENAME this as it's being used inside the contract
```

Comment:
Above this line we are using `createMessage` and are still passing in the `hash`. But is that right? Should be instead passing in the `id`? Or in addition to the `hash` also passing in the `id` to `createMessage`?

Also, there is the now outdated comment on lines 295 to 299. It should be rewritten to explain why `message.id()` is being used, instead of `.hash()` and the `TODO:` removed.

Reply:
Nice catch. Passing `hash` to the `createMessage` function is correct since it is used to create message.
But the problem is that in the chatroom contract, we should check if the hash is same.

Will remove TODO and update comment.